File: guides/micronaut-patterns-composite/java/src/main/java/example/micronaut/ColorController.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.micronaut.http.MediaType;
import java.util.Optional;

@Controller("/color") // <1>
public class ColorController {

    private final ColorFetcher colorFetcher;

    public ColorController(ColorFetcher colorFetcher) { // <2>
        this.colorFetcher = colorFetcher;
    }

    @Produces(MediaType.TEXT_PLAIN) // <3>
    @Get("/mint") // <4>
    Optional<String> mint(@NonNull HttpRequest<?> request) { // <5>
        return colorFetcher.favouriteColor(request);
    }

    @Produces(MediaType.TEXT_PLAIN) // <3>
    @Get
    Optional<String> index(@NonNull HttpRequest<?> request) { // <5>
        return colorFetcher.favouriteColor(request);
    }
}

File: guides/micronaut-patterns-composite/kotlin/src/main/kotlin/example/micronaut/ColorController.kt
package example.micronaut

import io.micronaut.http.HttpRequest
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import java.util.Optional

@Controller("/color") // <1>
class ColorController(private val colorFetcher: ColorFetcher) { // <2>

    @Produces(MediaType.TEXT_PLAIN) // <3>
    @Get("/mint") // <4>
    fun mint(request: HttpRequest<*>): Optional<String> = // <5>
        colorFetcher.favouriteColor(request)

    @Produces(MediaType.TEXT_PLAIN) // <3>
    @Get
    fun index(request: HttpRequest<*>): Optional<String> = // <5>
        colorFetcher.favouriteColor(request)
}
